### PR TITLE
fix: properly build xcframework

### PR DESCRIPTION
**Pull request by okwasniewski**
Hey, 

This PR features multiple changes to fix how `xcframeworks` are handled. 

Changes included: 
- Move from `macos-latest-large` to `macos-latest` for the CI (Only `build-dawn.yml` requires M1 machine for now)
- Remove visionOS x86 build (building on Intel macs for visionOS is deprecated so lets not waste resources on this)
- I aligned `apple.toolchain.cmake` with upstream source code.
- Package only `xcframeworks` in build dawn job (currently the artifacts were over 450mbs because contained unnecessary stuff)

### Changes to build scripts

So before we created a bunch of xcframeworks then linked them for each platform. 

The whole idea behind xcframeworks is to have one bundle for multiple platforms(!). 

Before (contents generated by build-dawn job): 

CleanShot 2024-09-20 at 14 36 12@2x

Now: 

CleanShot 2024-09-20 at 14 35 31@2x

### CODE REVIEW

**Comment by okwasniewski** [comment]
cc: @Saadnajmi @wcandillon

**Comment by wcandillon** [comment]
Looks great thank you :) :)

**Comment by okwasniewski** [comment]
Im not sure why this fails.. Probably we need to re-trigger build-dawn?

File: packages/webgpu/scripts/build/apple.toolchain.cmake [hunk 1]
```diff
@@ -67,10 +67,9 @@
 #    TVOSCOMBINED = Build for arm64 x86_64 tvOS + tvOS Simulator. Combined into FAT STATIC lib (only supported on 3.14+ of CMake with "-G Xcode" argument in combination with the "cmake --install" CMake build step)
 #    SIMULATOR_TVOS = Build for x86_64 tvOS Simulator.
 #    SIMULATORARM64_TVOS = Build for arm64 tvOS Simulator.
```
Comment (wcandillon) [inline packages/webgpu/scripts/build/apple.toolchain.cmake:69]: Could/should this be committed upstream to https://github.com/leetal/ios-cmake?

**Comment by okwasniewski** [comment]
@wcandillon The build is green! I missed that I removed additional live from podspec (which is now back)!

**Comment by wcandillon** [comment]
thanks for pushing the fixes :)

**Review by wcandillon — APPROVED** [review]

**Pull request merged**